### ISSUE TITLE
no-unused-vars false positive with typescript types

### ISSUE THREAD

**Issue by taylankasap** [babel/babel-eslint#835]
There are too many tools in the process so I'm not even sure which one's job this is. I'm just opening issue here for now. Please point me in the right direction. Thanks!

Typescript file (minimal example)
```typescript
import Rubik from './Rubik';

export default class Arranger {
  private readonly rubik: Rubik;
}
```

This is the false positive
```
error  'Rubik' is defined but never used  no-unused-vars
```

.eslintrc.json
```json
{
  "parser": "babel-eslint",
  "rules": {
    "no-unused-vars": ["error"]
  }
}
```

babel.config.json
```json
{
  "presets": [
    "@babel/preset-env",
    "@babel/preset-typescript"
  ],
  "plugins": [
    "@babel/proposal-class-properties"
  ]
}
```

package.json
```json
{
  "devDependencies": {
    "@babel/plugin-proposal-class-properties": "^7.8.3",
    "@babel/preset-typescript": "^7.9.0",
    "@symfony/webpack-encore": "^0.30.0",
    "babel-eslint": "^11.0.0-beta.2",
    "concurrently": "^5.2.0",
    "core-js": "^3.0.0",
    "eslint": "^7.0.0",
    "eslint-config-airbnb": "^18.1.0",
    "eslint-loader": "^4.0.2",
    "eslint-plugin-import": "^2.20.2",
    "eslint-plugin-tsdoc": "^0.2.5",
    "regenerator-runtime": "^0.13.2",
    "typescript": "^3.9.3",
    "webpack-notifier": "^1.6.0"
  },
  // ...
}
```

**Comment by balasan** [comment]
Can confirm - switch to `11.0.0-beta.2` to fix the `readonly` errors and ran into this one.

**Comment by kaicataldo** [comment]
Thank you for the PR. Now that [`@babel/eslint-parser` has been released](https://babeljs.io/blog/2020/07/13/the-state-of-babel-eslint), we are making this repository read-only. If this is a change you would still like to advocate for, please reopen this in the [babel/babel monorepo](https://github.com/babel/babel).
